pkg/virtctl/sev/attestation_test.go

```diff
@@ -0,0 +1,289 @@
+	var ctrl *gomock.Controller
+	var workDir string
+	var err error
+
+	BeforeEach(func() {
+		vm = kubecli.NewMinimalVM(vmiName)
+		vmi = api.NewMinimalVMI(vmiName)
+		ctrl = gomock.NewController(GinkgoT())
+		workDir, err = ioutil.TempDir("", "sev-test")
+		Expect(err).To(BeNil())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+		vmiInterface = kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
+		kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineInstance(k8smetav1.NamespaceDefault).Return(vmiInterface).AnyTimes()
+
+		vmiInterface.EXPECT().Get(vmi.Name, gomock.Any()).Return(vmi, nil).AnyTimes()
+		vmInterface.EXPECT().Get(vmi.Name, gomock.Any()).Return(vm, nil).AnyTimes()
+	})
+
+	AfterEach(func() {
+		if workDir != "" {
+			err = os.RemoveAll(workDir)
+			Expect(err).ToNot(HaveOccurred())
+			workDir = ""
+		}
+	})
+
+	Describe("The base 'sev' command", func() {
+		Context("Creation", func() {
+			It("should succeed", func() {
+				cmd := tests.NewVirtctlCommand(sev.COMMAND_SEV)
+				Expect(cmd).ToNot(BeNil())
+			})
+		})
+		Context("Running with no subcommand", func() {
+			It("should fail", func() {
+				cmd := tests.NewRepeatableVirtctlCommand(sev.COMMAND_SEV)
+				Expect(cmd()).ToNot(BeNil())
+			})
+		})
+	})
+
+	Describe("The 'fetch-cert-chain' command", func() {
+		Context("Creation", func() {
+			It("should succeed", func() {
+				cmd := tests.NewVirtctlCommand(sev.COMMAND_SEV, sev.COMMAND_FETCH_CERT_CHAIN)
+				Expect(cmd).ToNot(BeNil())
+			})
+		})
+		Context("With no argument", func() {
+			It("should fail", func() {
+				cmd := tests.NewVirtctlCommand(sev.COMMAND_SEV, sev.COMMAND_FETCH_CERT_CHAIN)
+				Expect(cmd.Execute()).ToNot(BeNil())
+			})
+		})
+		Context("With VMI name", func() {
+			It("should succeed", func() {
+				var sevPlatformInfo v1.SEVPlatformInfo
+				kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineInstance(k8smetav1.NamespaceDefault).Return(vmiInterface).Times(1)
+				vmiInterface.EXPECT().SEVFetchCertChain(vmi.Name).Return(sevPlatformInfo, nil).Times(1)
+
+				cmd := tests.NewVirtctlCommand(sev.COMMAND_SEV, sev.COMMAND_FETCH_CERT_CHAIN, vmiName)
+				Expect(cmd.Execute()).To(BeNil())
+			})
+			It("should print cert chain to file", func() {
+				sevPlatformInfo := v1.SEVPlatformInfo{
+					PDH:       "somelongpdh",
+					CertChain: "somelongcertchain",
+				}
+				fpath := filepath.Join(workDir, "cert_chain.json")
+				kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachineInstance(k8smetav1.NamespaceDefault).Return(vmiInterface).Times(1)
+				vmiInterface.EXPECT().SEVFetchCertChain(vmi.Name).Return(sevPlatformInfo, nil).Times(1)
+
+				cmd := tests.NewVirtctlCommand(sev.COMMAND_SEV, sev.COMMAND_FETCH_CERT_CHAIN, vmiName, "--output", fpath)
+				Expect(cmd.Execute()).To(BeNil())
+				// Expect file creation
```

Comment:
what about also checking that the content of the file matches `sevPlatformInfo` returned by the mock? Same for the measurement call.